### PR TITLE
SERVER-93485 Upgrade Dependencies

### PR DESCRIPTION
**Pull request by juangugit**
In this PR, I have update dependencies and minor changes

### CODE REVIEW

**Comment by juangugit** [comment]
### Cargo Update Output

```
(python3-venv) ubuntu@ip-10-128-38-2:~/mongo-task-generator$ cargo update
    Updating crates.io index
    Updating addr2line v0.17.0 -> v0.21.0
      Adding ahash v0.8.11
    Updating aho-corasick v0.7.18 -> v1.1.3
      Adding android-tzdata v0.1.1
      Adding android_system_properties v0.1.5
    Removing ansi_term v0.12.1
      Adding anstream v0.6.15
      Adding anstyle v1.0.8
      Adding anstyle-parse v0.2.5
      Adding anstyle-query v1.1.1
      Adding anstyle-wincon v3.0.4
    Updating anyhow v1.0.58 -> v1.0.86
    Removing arrayvec v0.5.2
    Updating assert_cmd v2.0.4 -> v2.0.16
      Adding async-scoped v0.9.0
    Updating async-trait v0.1.56 -> v0.1.81
      Adding atomic-waker v1.1.2
    Updating atomicwrites v0.3.1 -> v0.4.3
    Removing atty v0.2.14
    Updating autocfg v1.1.0 -> v1.3.0
    Updating backtrace v0.3.65 -> v0.3.71
      Adding backtrace-ext v0.2.1
    Removing base64 v0.13.0
      Adding base64 v0.21.7
      Adding base64 v0.22.1
      Adding base64ct v1.6.0
      Adding bitflags v2.6.0
      Adding block-buffer v0.10.4
    Updating bstr v0.2.17 -> v1.10.0
    Updating bumpalo v3.10.0 -> v3.16.0
    Removing bytecount v0.3.2
    Removing bytecount v0.6.3
      Adding byteorder v1.5.0
    Updating bytes v1.1.0 -> v1.7.1
    Updating camino v1.0.9 -> v1.1.7
      Adding camino-tempfile v1.1.1
    Updating cargo-nextest v0.9.23 -> v0.9.72
    Updating cargo-platform v0.1.2 -> v0.1.8
    Updating cargo_metadata v0.14.2 -> v0.18.1
    Updating cc v1.0.73 -> v1.1.10
    Updating cfg-expr v0.10.3 -> v0.16.0
      Adding cfg_aliases v0.1.1
    Updating chrono v0.4.19 -> v0.4.38
    Updating clap v3.2.7 -> v4.5.15
      Adding clap_builder v4.5.15
    Updating clap_derive v3.2.7 -> v4.5.13
    Updating clap_lex v0.2.4 -> v0.7.2
    Updating cmd_lib v1.3.0 -> v1.9.4
    Updating cmd_lib_macros v1.3.0 -> v1.9.4
    Updating color-eyre v0.6.1 -> v0.6.3
      Adding colorchoice v1.0.2
    Updating config v0.13.1 -> v0.14.0
    Updating console v0.15.0 -> v0.15.8
      Adding const-oid v0.9.6
    Updating core-foundation v0.9.3 -> v0.9.4
    Updating core-foundation-sys v0.8.3 -> v0.8.7
      Adding cpufeatures v0.2.13
    Updating crc32fast v1.3.2 -> v1.4.2
    Removing crossbeam-channel v0.5.5
    Removing crossbeam-deque v0.8.1
    Removing crossbeam-epoch v0.9.9
    Removing crossbeam-utils v0.8.10
      Adding crypto-common v0.1.6
    Removing ctrlc v3.2.2
      Adding curve25519-dalek v4.1.3
      Adding curve25519-dalek-derive v0.1.1
    Updating debug-ignore v1.0.2 -> v1.0.5
      Adding der v0.7.9
    Updating dialoguer v0.10.1 -> v0.11.0
      Adding digest v0.10.7
      Adding dirs v5.0.1
    Removing dirs-next v2.0.0
      Adding dirs-sys v0.4.1
    Removing dirs-sys-next v0.1.2
      Adding display-error-chain v0.2.0
    Updating duct v0.13.5 -> v0.13.7
      Adding dunce v1.0.5
      Adding ed25519 v2.2.3
      Adding ed25519-dalek v2.1.1
    Updating either v1.6.1 -> v1.13.0
    Updating enable-ansi-support v0.1.2 -> v0.2.1
    Updating encoding_rs v0.8.31 -> v0.8.34
      Adding env_filter v0.1.2
    Removing env_logger v0.9.0
      Adding env_logger v0.10.2
      Adding env_logger v0.11.5
      Adding equivalent v1.0.1
      Adding errno v0.3.9
    Updating eyre v0.6.8 -> v0.6.12
    Removing fastrand v1.7.0
      Adding fastrand v1.9.0
      Adding fastrand v2.1.0
      Adding fiat-crypto v0.2.9
    Updating filetime v0.2.16 -> v0.2.24
    Updating fixedbitset v0.4.1 -> v0.4.2
    Updating flate2 v1.0.24 -> v1.0.31
    Updating form_urlencoded v1.0.1 -> v1.2.1
      Adding future-queue v0.3.0
    Updating futures v0.3.21 -> v0.3.30
    Updating futures-channel v0.3.21 -> v0.3.30
    Updating futures-core v0.3.21 -> v0.3.30
    Updating futures-executor v0.3.21 -> v0.3.30
    Updating futures-io v0.3.21 -> v0.3.30
    Updating futures-macro v0.3.21 -> v0.3.30
    Updating futures-sink v0.3.21 -> v0.3.30
    Updating futures-task v0.3.21 -> v0.3.30
    Updating futures-timer v3.0.2 -> v3.0.3
    Updating futures-util v0.3.21 -> v0.3.30
      Adding generic-array v0.14.7
    Updating getrandom v0.2.7 -> v0.2.15
    Updating gimli v0.26.1 -> v0.28.1
      Adding glob v0.3.1
      Adding globset v0.4.14
    Updating guppy v0.14.2 -> v0.17.7
    Removing h2 v0.3.13
      Adding h2 v0.3.26
      Adding h2 v0.4.5
    Removing hashbrown v0.12.1
      Adding hashbrown v0.12.3
      Adding hashbrown v0.14.5
    Updating heck v0.4.0 -> v0.5.0
    Updating hermit-abi v0.1.19 -> v0.3.9
    Updating home v0.5.3 -> v0.5.9
    Removing http v0.2.8
      Adding http v0.2.12
      Adding http v1.1.0
    Removing http-body v0.4.5
      Adding http-body v0.4.6
      Adding http-body v1.0.1
      Adding http-body-util v0.1.2
    Updating httparse v1.7.1 -> v1.9.4
    Updating httpdate v1.0.2 -> v1.0.3
    Removing hyper v0.14.19
      Adding hyper v0.14.30
      Adding hyper v1.4.1
    Removing hyper-rustls v0.23.0
      Adding hyper-rustls v0.24.2
      Adding hyper-rustls v0.27.2
      Adding hyper-tls v0.6.0
      Adding hyper-util v0.1.7
      Adding iana-time-zone v0.1.60
      Adding iana-time-zone-haiku v0.1.2
    Updating idna v0.2.3 -> v0.5.0
    Removing indent_write v2.2.0
    Removing indexmap v1.9.1
      Adding indexmap v1.9.3
      Adding indexmap v2.3.0
    Updating indicatif v0.16.2 -> v0.17.8
    Updating instant v0.1.12 -> v0.1.13
    Updating ipnet v2.5.0 -> v2.9.0
      Adding is-terminal v0.4.12
    Updating is_ci v1.1.1 -> v1.2.0
      Adding is_terminal_polyfill v1.70.1
    Removing itertools v0.10.3
      Adding itertools v0.12.1
      Adding itertools v0.13.0
    Updating itoa v1.0.2 -> v1.0.11
    Updating jobserver v0.1.24 -> v0.1.32
    Updating js-sys v0.3.58 -> v0.3.69
    Updating lazy_static v1.4.0 -> v1.5.0
    Removing lexical-core v0.7.6
    Updating libc v0.2.126 -> v0.2.155
      Adding libredox v0.1.3
      Adding linux-raw-sys v0.4.14
    Updating lock_api v0.4.7 -> v0.4.12
    Updating log v0.4.17 -> v0.4.22
    Removing matches v0.1.9
    Updating memchr v2.5.0 -> v2.7.4
    Removing memoffset v0.6.5
    Updating miette v4.7.1 -> v7.2.0
    Updating miette-derive v4.7.1 -> v7.2.0
    Updating mime v0.3.16 -> v0.3.17
    Removing mime_guess v2.0.4
    Updating miniz_oxide v0.5.3 -> v0.7.4
    Updating mio v0.8.4 -> v1.0.2
    Updating mongo-task-generator v0.7.15 (/home/ubuntu/mongo-task-generator) -> v0.7.16
    Updating mukti-metadata v0.1.0 -> v0.2.1
    Updating native-tls v0.2.11 -> v0.2.12
      Adding newtype-uuid v1.1.0
    Updating nextest-filtering v0.1.0 -> v0.9.0
    Updating nextest-metadata v0.4.3 -> v0.11.0
    Updating nextest-runner v0.14.0 -> v0.56.1
      Adding nextest-workspace-hack v0.1.0
    Updating nix v0.24.1 -> v0.28.0
    Removing nom v5.1.2
    Removing nom v7.1.1
      Adding nom v7.1.3
    Removing nom-tracable v0.8.0
    Removing nom-tracable-macros v0.8.0
    Removing nom_locate v1.0.0
    Removing nom_locate v4.0.0
      Adding nu-ansi-term v0.46.0
    Removing num-integer v0.1.45
    Updating num-traits v0.2.15 -> v0.2.19
    Removing num_cpus v1.13.1
    Updating object v0.28.4 -> v0.32.2
    Updating once_cell v1.12.0 -> v1.19.0
    Updating openssl v0.10.42 -> v0.10.66
    Updating openssl-macros v0.1.0 -> v0.1.1
    Updating openssl-sys v0.9.77 -> v0.9.103
      Adding option-ext v0.2.0
    Updating os_pipe v0.9.2 -> v1.2.1
    Removing os_str_bytes v6.1.0
      Adding overload v0.1.1
    Removing owo-colors v3.4.0
      Adding owo-colors v3.5.0
      Adding owo-colors v4.0.0
    Removing parking_lot v0.12.1
      Adding parking_lot v0.11.2
      Adding parking_lot v0.12.3
    Removing parking_lot_core v0.9.3
      Adding parking_lot_core v0.8.6
      Adding parking_lot_core v0.9.10
    Updating percent-encoding v2.1.0 -> v2.3.1
    Updating petgraph v0.6.2 -> v0.6.5
      Adding pin-project v1.1.5
      Adding pin-project-internal v1.1.5
    Updating pin-project-lite v0.2.9 -> v0.2.14
      Adding pkcs8 v0.10.2
    Updating pkg-config v0.3.26 -> v0.3.30
      Adding portable-atomic v1.7.0
    Updating ppv-lite86 v0.2.16 -> v0.2.20
    Updating predicates v2.1.1 -> v3.1.2
    Updating predicates-core v1.0.3 -> v1.0.8
    Updating predicates-tree v1.0.5 -> v1.0.11
      Adding proc-macro-crate v3.1.0
    Updating proc-macro2 v1.0.40 -> v1.0.86
    Updating quick-junit v0.2.0 -> v0.4.0
    Removing quick-xml v0.22.0
    Removing quick-xml v0.23.0
      Adding quick-xml v0.23.1
      Adding quick-xml v0.31.0
    Updating quote v1.0.20 -> v1.0.36
    Updating rand_core v0.6.3 -> v0.6.4
    Removing rayon v1.5.3
    Removing rayon-core v1.9.3
      Adding recursion v0.5.2
    Removing redox_syscall v0.2.13
      Adding redox_syscall v0.2.16
      Adding redox_syscall v0.5.3
    Updating redox_users v0.4.3 -> v0.4.5
    Updating regex v1.5.6 -> v1.10.6
    Updating regex-automata v0.1.10 -> v0.4.7
    Updating regex-syntax v0.6.26 -> v0.8.4
      Adding relative-path v1.9.3
    Removing reqwest v0.11.13
      Adding reqwest v0.11.27
      Adding reqwest v0.12.5
    Updating reqwest-middleware v0.2.0 -> v0.3.3
    Updating reqwest-retry v0.2.0 -> v0.6.1
    Updating retry-policies v0.1.2 -> v0.4.0
    Updating ring v0.16.20 -> v0.17.8
      Adding ron v0.8.1
    Updating rstest v0.15.0 -> v0.22.0
    Updating rstest_macros v0.14.0 -> v0.22.0
    Updating rustc-demangle v0.1.21 -> v0.1.24
      Adding rustix v0.38.34
    Removing rustls v0.20.6
      Adding rustls v0.21.12
      Adding rustls v0.23.12
    Removing rustls-pemfile v1.0.0
      Adding rustls-pemfile v1.0.4
      Adding rustls-pemfile v2.1.3
      Adding rustls-pki-types v1.8.0
      Adding rustls-webpki v0.101.7
      Adding rustls-webpki v0.102.6
    Updating ryu v1.0.10 -> v1.0.18
    Updating schannel v0.1.20 -> v0.1.23
    Updating scopeguard v1.1.0 -> v1.2.0
    Updating sct v0.7.0 -> v0.7.1
    Updating security-framework v2.7.0 -> v2.11.1
    Updating security-framework-sys v2.6.1 -> v2.11.1
      Adding self-replace v1.4.0
    Updating self_update v0.30.0 -> v0.39.0
    Updating semver v1.0.10 -> v1.0.23
    Updating serde v1.0.137 -> v1.0.207
    Updating serde_derive v1.0.137 -> v1.0.207
      Adding serde_ignored v0.1.10
    Updating serde_json v1.0.81 -> v1.0.124
      Adding serde_path_to_error v0.1.16
      Adding serde_spanned v0.6.7
    Removing serde_yaml v0.8.24
      Adding serde_yaml v0.8.26
      Adding serde_yaml v0.9.34+deprecated
      Adding sha2 v0.10.8
    Updating sharded-slab v0.1.4 -> v0.1.7
    Updating shared_child v0.3.5 -> v1.0.1
    Updating shellexpand v2.1.0 -> v3.1.0
    Updating signal-hook-registry v1.4.0 -> v1.4.2
      Adding signature v2.2.0
      Adding simple-error v0.3.1
    Updating slab v0.4.6 -> v0.4.9
    Updating smallvec v1.8.1 -> v1.13.2
    Updating smawk v0.3.1 -> v0.3.2
      Adding smol_str v0.2.2
    Updating socket2 v0.4.4 -> v0.5.7
    Updating spin v0.5.2 -> v0.9.8
      Adding spki v0.7.3
    Updating strip-ansi-escapes v0.1.1 -> v0.2.0
    Updating strsim v0.10.0 -> v0.11.1
      Adding subtle v2.6.1
    Removing supports-color v1.3.0
      Adding supports-color v2.1.0
      Adding supports-color v3.0.0
    Updating supports-hyperlinks v1.2.0 -> v3.0.0
    Updating supports-unicode v1.0.2 -> v3.0.0
      Adding swrite v0.1.0
    Removing syn v1.0.98
      Adding syn v1.0.109
      Adding syn v2.0.74
      Adding sync_wrapper v0.1.2
      Adding sync_wrapper v1.0.1
      Adding system-configuration v0.5.1
      Adding system-configuration-sys v0.5.0
    Updating tar v0.4.38 -> v0.4.41
    Updating target-lexicon v0.12.4 -> v0.12.16
    Updating target-spec v1.0.2 -> v3.2.1
      Adding target-spec-miette v0.4.0
    Removing task-local-extensions v0.1.3
    Updating tempfile v3.3.0 -> v3.12.0
    Updating termcolor v1.1.3 -> v1.4.1
    Updating terminal_size v0.1.17 -> v0.3.0
    Updating termtree v0.2.4 -> v0.4.1
    Updating textwrap v0.15.0 -> v0.16.1
    Updating thiserror v1.0.31 -> v1.0.63
    Updating thiserror-impl v1.0.31 -> v1.0.63
    Updating thread_local v1.1.4 -> v1.1.8
    Removing time v0.1.44
    Updating tinyvec v1.6.0 -> v1.8.0
    Updating tinyvec_macros v0.1.0 -> v0.1.1
    Updating tokio v1.19.2 -> v1.39.2
    Updating tokio-macros v1.8.0 -> v2.4.0
    Updating tokio-native-tls v0.3.0 -> v0.3.1
    Removing tokio-rustls v0.23.4
      Adding tokio-rustls v0.24.1
      Adding tokio-rustls v0.26.0
    Updating tokio-util v0.7.3 -> v0.7.11
    Updating toml v0.5.9 -> v0.8.19
      Adding toml_datetime v0.6.8
      Adding toml_edit v0.21.1
      Adding toml_edit v0.22.20
      Adding tower v0.4.13
      Adding tower-layer v0.3.2
    Updating tracing v0.1.35 -> v0.1.40
    Updating tracing-attributes v0.1.21 -> v0.1.27
    Updating tracing-core v0.1.28 -> v0.1.32
    Updating tracing-log v0.1.3 -> v0.2.0
    Updating tracing-subscriber v0.3.11 -> v0.3.18
    Updating try-lock v0.2.3 -> v0.2.5
    Removing twox-hash v1.6.3
      Adding typenum v1.17.0
    Removing unicase v2.6.0
    Updating unicode-bidi v0.3.8 -> v0.3.15
    Updating unicode-ident v1.0.1 -> v1.0.12
    Updating unicode-linebreak v0.1.2 -> v0.1.5
    Updating unicode-normalization v0.1.20 -> v0.1.23
    Updating unicode-width v0.1.9 -> v0.1.13
      Adding unsafe-libyaml v0.2.11
    Updating untrusted v0.7.1 -> v0.9.0
    Updating url v2.2.2 -> v2.5.2
      Adding urlencoding v2.1.3
    Updating utf8parse v0.2.0 -> v0.2.2
      Adding uuid v1.10.0
    Updating version_check v0.9.4 -> v0.9.5
    Updating vte v0.10.1 -> v0.11.1
    Updating vte_generate_state_changes v0.1.1 -> v0.1.2
    Updating want v0.3.0 -> v0.3.1
    Removing wasi v0.10.0+wasi-snapshot-preview1
    Updating wasm-bindgen v0.2.81 -> v0.2.92
    Updating wasm-bindgen-backend v0.2.81 -> v0.2.92
    Updating wasm-bindgen-futures v0.4.31 -> v0.4.42
    Updating wasm-bindgen-macro v0.2.81 -> v0.2.92
    Updating wasm-bindgen-macro-support v0.2.81 -> v0.2.92
    Updating wasm-bindgen-shared v0.2.81 -> v0.2.92
      Adding wasm-timer v0.2.5
    Updating web-sys v0.3.58 -> v0.3.69
    Removing webpki v0.22.0
    Updating webpki-roots v0.22.3 -> v0.25.4
      Adding win32job v2.0.0
    Updating winapi-util v0.1.5 -> v0.1.9
    Updating windows v0.38.0 -> v0.52.0
      Adding windows-core v0.52.0
    Removing windows-sys v0.36.1
      Adding windows-sys v0.42.0
      Adding windows-sys v0.48.0
      Adding windows-sys v0.52.0
      Adding windows-sys v0.59.0
      Adding windows-targets v0.48.5
      Adding windows-targets v0.52.6
      Adding windows_aarch64_gnullvm v0.42.2
      Adding windows_aarch64_gnullvm v0.48.5
      Adding windows_aarch64_gnullvm v0.52.6
    Removing windows_aarch64_msvc v0.36.1
    Removing windows_aarch64_msvc v0.38.0
      Adding windows_aarch64_msvc v0.42.2
      Adding windows_aarch64_msvc v0.48.5
      Adding windows_aarch64_msvc v0.52.6
    Removing windows_i686_gnu v0.36.1
    Removing windows_i686_gnu v0.38.0
      Adding windows_i686_gnu v0.42.2
      Adding windows_i686_gnu v0.48.5
      Adding windows_i686_gnu v0.52.6
      Adding windows_i686_gnullvm v0.52.6
    Removing windows_i686_msvc v0.36.1
    Removing windows_i686_msvc v0.38.0
      Adding windows_i686_msvc v0.42.2
      Adding windows_i686_msvc v0.48.5
      Adding windows_i686_msvc v0.52.6
    Removing windows_x86_64_gnu v0.36.1
    Removing windows_x86_64_gnu v0.38.0
      Adding windows_x86_64_gnu v0.42.2
      Adding windows_x86_64_gnu v0.48.5
      Adding windows_x86_64_gnu v0.52.6
      Adding windows_x86_64_gnullvm v0.42.2
      Adding windows_x86_64_gnullvm v0.48.5
      Adding windows_x86_64_gnullvm v0.52.6
    Removing windows_x86_64_msvc v0.36.1
    Removing windows_x86_64_msvc v0.38.0
      Adding windows_x86_64_msvc v0.42.2
      Adding windows_x86_64_msvc v0.48.5
      Adding windows_x86_64_msvc v0.52.6
      Adding winnow v0.5.40
      Adding winnow v0.6.18
    Removing winreg v0.10.1
      Adding winreg v0.50.0
      Adding winreg v0.52.0
    Updating xattr v0.2.3 -> v1.3.1
      Adding xxhash-rust v0.8.12
      Adding zerocopy v0.7.35
      Adding zerocopy-derive v0.7.35
    Updating zeroize v1.5.5 -> v1.8.1
      Adding zipsign-api v0.1.2
    Updating zstd v0.11.2+zstd.1.5.2 -> v0.13.2
    Updating zstd-safe v5.0.2+zstd.1.5.2 -> v7.2.1
    Updating zstd-sys v2.0.1+zstd.1.5.2 -> v2.0.13+zstd.1.5.6
```

File: tests/integration_test.rs
Comment (juangugit) [inline tests/integration_test.rs:1]: The switch to raw string literals (`r#"..."#`) is related to fixing errors caused by the Rust upgrade.

File: src/main.rs
Comment (juangugit) [inline src/main.rs:1]: The change from `parse(from_os_str)` to `value_parser` is related to the `clap` crate upgrade.

**Review by Trevor159 — APPROVED** [review]
lgtm but I would wait for review from @MikhailShchatko

File: Cargo.toml
Comment (smcclure15) [inline Cargo.toml:11]: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements

"Leaving off the caret is a simplified equivalent syntax to using caret requirements. While caret requirements are the default, it is recommended to use the simplified syntax when possible."

**Review by zackwintermdb — APPROVED** [review]
LGTM

**Pull request merged**
